src/lib/y2network/clients/routing.rb

```diff
@@ -0,0 +1,466 @@
+
+          CommandLine.Print(
+            Builtins.sformat(
+              _("Adding '%1' destination to routing table ..."),
+              destination
+            )
+          )
+          yast_routing_table.routes << route if !yast_routing_table.routes.include?(route)
+        when :edit
+          if destination == ""
+            CommandLine.Error(_("Destination IP address must be specified."))
+            return false
+          end
+          if options.size < 2
+            CommandLine.Error(
+              _(
+                "At least one of the following parameters (gateway, netmask, device, options) must be specified"
+              )
+            )
+            return false
+          end
+
+          route = find_route_by_destination(destination)
+          current_routes.find { |r| serializer.to_hash(r)["destination"] == destination }
+          edited_route = route_from_options(m)
+
+          if route
+            CommandLine.Print(
+              Builtins.sformat(
+                _("Updating '%1' destination in routing table ..."),
+                destination
+              )
+            )
+
+            update_route_with!(route, edited_route)
+          else
+            CommandLine.Error(
+              Builtins.sformat(
+                _("No entry for destination '%1' in routing table"),
+                destination
+              )
+            )
+            return false
+          end
+        end
+
+        true
+      end
+
+      def AddHandler(options)
+        AddEditHandler(:add, options)
+        true
+      end
+
+      def EditHandler(options)
+        AddEditHandler(:edit, options)
+        true
+      end
+
+      def DeleteHandler(options)
+        destination = options["dest"]
+        route = find_route_by_destination(destination)
+
+        if route
+          CommandLine.Print(
+            Builtins.sformat(
+              _("Deleting '%1' destination from routing table ..."),
+              destination
+            )
+          )
+          yast_routing_table.routes.delete(route)
+        else
+          CommandLine.Error(
+            Builtins.sformat(
+              _("No entry for destination '%1' in routing table"),
+              destination
+            )
+          )
+        end
+
+        !!route
+      end
+
+      def read
+        Yast::Lan.add_config(:yast, Y2Network::Config.from(:sysconfig))
+        Yast::Lan.add_config(:system, Y2Network::Config.from(:sysconfig))
+
+        true
+      end
+
+      def write
+        # TODO: Do not write if not modified
+        if modified?
+          Y2Network::ConfigWriter::Sysconfig.new.write(yast_config)
+          log.info("Writing routing configuration: #{yast_config.routing.inspect}")
+        end
+
+        true
+      end
+
+      def modified?
+        yast_config.routing != Yast::Lan.system_config.routing
+      end
+
+      def yast_config
+        Yast::Lan.yast_config
+      end
+
+      def yast_routing_table
```

Comment:
now I see that collision. I agree that include for such big think as routines.rb was bad idea and hopefully will die soon.